### PR TITLE
Add DLC storage implementation

### PR DESCRIPTION
**Pull request by benthecarman**
implement DLC storage traits for browser storage

### CODE REVIEW

File: node-manager/src/dlcstorage.rs [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+
+    fn update_contract(&mut self, contract: &Contract) -> Result<(), Error> {
+        let serialized = serialize_contract(contract)?;
+        match contract {
+            a @ Contract::Accepted(_) | a @ Contract::Signed(_) => {
+                let key = format! {"{DLC_CONTRACT_KEY_PREFIX}{}", a.get_temporary_id().to_hex()};
+                MutinyBrowserStorage::delete(key);
+            }
+            _ => {}
+        };
```
Comment (TonyGiorgio) [inline node-manager/src/dlcstorage.rs:69]: Should you delete after `set` instead? What if there's a problem saving the contract?

File: node-manager/src/dlcstorage.rs [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+        let result: Result<Vec<u8>, MutinyStorageError> = self.get(DLC_CHAIN_MONITOR_KEY);
+
+        match result {
+            Ok(bytes) => {
+                let mon = ChainMonitor::deserialize(&mut Cursor::new(bytes)).map_err(|_| {
+                    Error::StorageError("Failed to decode ChainMonitor".to_string())
+                })?;
+                Ok(Some(mon))
+            }
+            Err(_) => Ok(None),
```
Comment (TonyGiorgio) [inline node-manager/src/dlcstorage.rs:189]: Could you match on error type here? To distinguish a problem getting the value vs if a value does not currently exist?

File: node-manager/src/localstorage.rs [hunk 3]
```diff
@@ -85,6 +87,36 @@ impl MutinyBrowserStorage {
         map
     }
 
+    pub(crate) fn get_data_with_prefix<T: Serializable>(
```
Comment (TonyGiorgio) [inline node-manager/src/localstorage.rs:90]: how is this different than `scan`?

Comment (benthecarman) [reply node-manager/src/localstorage.rs:90]: scan checks the key, this checks the value

**Review by TonyGiorgio — APPROVED** [review]

**Pull request merged**